components/movementsensor/rtkutils/cachedData.go

```diff
@@ -0,0 +1,176 @@
+package rtkutils
+
+import (
+	"context"
+	"math"
+	"sync"
+
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/spatialmath"
+)
+
+var errNilLocation = errors.New("nil gps location, check nmea message parsing")
+
+// CachedGpsData allows the use of any MovementSensor chip via a DataReader.
+type CachedGpsData struct {
```

Comment:
instead of "CachedGpsData" can we call it "CachedData"?

Reply:
I want the name to show that it's related to the `GPSData` struct. and TBH I kinda want to rename that one, too, but I'm not sure yet. Would `CachedData` and `Data` be alright?

Reply:
...no, maybe they're separate enough that the names don't need to be linked. I renamed `GPSData` to `NmeaParser` to match its file name, and will rename `CachedGpsData` to just `CachedData` as you suggested. Thanks for the idea!